src/gui/transferlistmodel.h

```diff
@@ -39,6 +40,47 @@ namespace BitTorrent
     class TorrentHandle;
 }
 
+// convenience class to read text colors from qss since qss doesn't work with qobject
+class TransferListTextColors : public QWidget
+{
+    Q_OBJECT
+    Q_DISABLE_COPY(TransferListTextColors);
+
+public:
+    Q_PROPERTY(QColor unknownTextColor MEMBER m_unknownTextColor);
+    Q_PROPERTY(QColor forcedDownloadingTextColor MEMBER m_forcedDownloadingTextColor);
+    Q_PROPERTY(QColor downloadingTextColor MEMBER m_downloadingTextColor);
+    Q_PROPERTY(QColor downloadingMetadataTextColor MEMBER m_downloadingMetadataTextColor);
+    Q_PROPERTY(QColor allocatingTextColor MEMBER m_allocatingTextColor);
+    Q_PROPERTY(QColor stalledDownloadingTextColor MEMBER m_stalledDownloadingTextColor);
+    Q_PROPERTY(QColor forcedUploadingTextColor MEMBER m_forcedUploadingTextColor);
+    Q_PROPERTY(QColor uploadingTextColor MEMBER m_uploadingTextColor);
+    Q_PROPERTY(QColor stalledUploadingTextColor MEMBER m_stalledUploadingTextColor);
+    Q_PROPERTY(QColor checkingResumeDataTextColor MEMBER m_checkingResumeDataTextColor);
+    Q_PROPERTY(QColor queuedDownloadingTextColor MEMBER m_queuedDownloadingTextColor);
+    Q_PROPERTY(QColor queuedUploadingTextColor MEMBER m_queuedUploadingTextColor);
+    Q_PROPERTY(QColor checkingUploadingTextColor MEMBER m_checkingUploadingTextColor);
+    Q_PROPERTY(QColor checkingDownloadingTextColor MEMBER m_checkingDownloadingTextColor);
+    Q_PROPERTY(QColor pausedDownloadingTextColor MEMBER m_pausedDownloadingTextColor);
+    Q_PROPERTY(QColor pausedUploadingTextColor MEMBER m_pausedUploadingTextColor);
+    Q_PROPERTY(QColor movingTextColor MEMBER m_movingTextColor);
+    Q_PROPERTY(QColor missingFilesTextColor MEMBER m_missingFilesTextColor);
+    Q_PROPERTY(QColor errorTextColor MEMBER m_errorTextColor);
+
+    TransferListTextColors();
+
+    QColor textColorByState(int) const;
```

Comment:
you missed giving the function argument a name.

Reply:
done